src/drivers/magnetometer/ak09916/ak09916.cpp

```diff
@@ -583,8 +583,7 @@ AK09916::write_reg(uint8_t reg, uint8_t value)
 int
 AK09916::reset(void)
 {
-	// First initialize it to use the bus
-	int rv = setup();
+	int rv = probe();
```

Comment:
I would remove the reset logic from probe().

Reply:
I would too but I wanted to make minimal changes.